### DISCUSSION TITLE
How to add span links

### DISCUSSION THREAD

**Discussion by guadalupe-duolingo** [open-telemetry/opentelemetry-java-instrumentation#8860] [Q&A]
I've been trying to add span links to trace a span through a SQS queue but I can't find how to do this. I found [this documentation ](https://opentelemetry.io/docs/concepts/signals/traces/#span-links) that explains what span links are but I can't find any documentation that details what code is necessary to implement this. Are there any examples I can look at to see how this can de done?

**Comment by laurit** [comment]
Span links are added through `SpanBuilder` see https://github.com/open-telemetry/opentelemetry-java/blob/749d3164cce52a81a5d52882c4a5f716cfd2f3a3/api/all/src/main/java/io/opentelemetry/api/trace/SpanBuilder.java#L143-L172

**Reply by guadalupe-duolingo** [reply]
Thanks.  How I would use the functions in the `SpanBuilder` interface, would I need to do this: `TracerProvider.noop().tracerBuilder("instrumentationScopeName").build()` or is there a simpler way to do it?
Also, since I'm trying to link messages that go through a queue asynchronously would this require passing the SpanContext when the message is put into the queue and then making the link when the message is pulled out of the queue with the SpanContext that was passed in?
